docs: add dns setup

noahpb opened the pull request:
## Description
Adding documentation on the DNS assumptions in UDS Core and added example for deploying in a non-dev environment.

## Related Issue

Fixes https://github.com/defenseunicorns/uds-core/issues/730

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Other (security config, docs update, etc)

## Checklist before merging

- [ ] Test, docs, adr added or updated as needed
- [ ] [Contributor Guide](https://github.com/defenseunicorns/uds-template-capability/blob/main/CONTRIBUTING.md) followed

noahpb committed:
docs: reflect that we are no longer using haproxy in uds-k3d-* bundles

noahpb committed:
docs: inital commit for dns

noahpb committed:
docs update

noahpb committed:
rm section

noahpb committed:
Merge branch 'main' into docs/add-dns

noahpb committed:
lint fix

mjnagel commented on docs/deployment/dns.md:
```diff
@@ -0,0 +1,68 @@
+- *.admin.<DOMAIN> / Admin Gateway
+
+{{% alert-note %}}
+The default value for `DOMAIN` is `uds.dev`, which is intended for development purposes only. For non-development purposes, you should override this value by specifying a value for `domain` in your `uds-config.yaml`. You can read more about overriding variables [here](https://uds.defenseunicorns.com/cli/quickstart-and-usage/#variables-and-configuration). 
```
Could also link the specific domain/cert config doc: https://uds.defenseunicorns.com/core/configuration/istio/ingress/#configure-domain-name-and-tls-for-istio-gateways

mjnagel commented on docs/deployment/dns.md:
```diff
@@ -0,0 +1,68 @@
+These service annotations and their values are subject to change. Please reference documentation from your cloud provider to ensure their validity.
+{{% /alert-note %}}
+
+From here, you may create DNS records to associate each LoadBalancer with its appropriate domain name.
```
I think it might be helpful to give an example of how to pull the LB CNAME/IPs from the cluster (`k get svc ...` and look at the external ip/host), and then reference setting up a CNAME record for each of the wildcards to point to the host given.

mjnagel commented on docs/deployment/dns.md:
````diff
@@ -0,0 +1,68 @@
+```
+{{% alert-note %}}
+These service annotations and their values are subject to change. Please reference documentation from your cloud provider to ensure their validity.
+{{% /alert-note %}}
````
Not sure if this is the right place or necessary, but thoughts on adding a note about static IPs to reduce effects of LBs cycling? Maybe too cloud provider specific though...

noahpb replied:
That's great callout and a good best-practice. I'm not sure if we want to get into cloud-specific deployment best practices in this document though. Unless we disagree that most folks reading this documentation are aware of this behavior, I'm leaning towards keeping this out.

noahpb replied:
Maybe we should just remove this callout altogether, it feels unnecessary the more I think about it.

mjnagel replied:
Yeah might be fine to just delete this note - since it's just meant to be an example anyways?

noahpb committed:
feedback

noahpb committed:
nit: wording

noahpb committed:
change wording

noahpb committed:
better wording

noahpb committed:
Merge branch 'main' into docs/add-dns

mjnagel commented on docs/deployment/dns.md:
```diff
@@ -0,0 +1,77 @@
+### Domain Configuration
+Each Gateway is associated to a wildcard DNS entry that is derived from the `DOMAIN` [variable](https://github.com/defenseunicorns/uds-core/blob/e624d73f79bd6739b6808fbdbf5ca75ebb7c1d3c/src/istio/zarf.yaml#L8) in the UDS Core Istio package. When deploying UDS Core, you can expect two Gateways to be created that match the following domain names:
+- *.<DOMAIN> / Tenant Gateway
+- *.admin.<DOMAIN> / Admin Gateway
```
![Screenshot 2024-09-19 at 8 13 12 AM](https://github.com/user-attachments/assets/07890787-9bd6-4401-a022-b8d528e21a8d)

These get clobbered on the docs site preview - could either escape the `<` and `>` characters, or just wrap in code backticks.

noahpb replied:
Oops! how did you render this out?

mjnagel replied:
Using the [docs repo](https://github.com/defenseunicorns/uds-docs), if you have `hugo` installed locally you can just run `hugo serve`. You will want to clone or copy core into the `./repo-docs` folder first (I use a symlink locally for this), see the script we use to manage this: https://github.com/defenseunicorns/uds-docs/blob/main/scripts/integrated-docs.sh#L22

mjnagel replied:
Would love to better integrate/task-ify this flow: https://github.com/defenseunicorns/uds-core/issues/712

noahpb replied:
Updated:
![image](https://github.com/user-attachments/assets/0862e359-0b3b-4624-ba97-267264a72d05)

mjnagel commented on docs/deployment/dns.md:
````diff
@@ -0,0 +1,77 @@
+istio-tenant-gateway        tenant-ingressgateway                            LoadBalancer   10.43.47.182    k8s-istioten-tenant...elb.us-east-1.amazonaws.com  15021:31222/TCP,80:30456/TCP,443:32508/TCP  1h
+```
+
+From here, DNS records can be created for the environment. You can create A records that associate your top-level domain to the Public IP Addresses of your Load Balancers. For subdomains, CNAME records can be used. For more information on how to configure DNS in AWS, see this [documentation](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/routing-to-elb-load-balancer.html).
````
> For more information on how to configure DNS in AWS, see this [documentation](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/routing-to-elb-load-balancer.html).

I think we can leave it if you think it would be valuable, but want to be careful about giving the impression that this only works/has been tested with AWS. Unfortunately I think most docs we could link would be provider specific (i.e. Route53, Cloudflare, ...). Maybe sufficient to just drop the last line and reference "refer to your DNS provider's documentation"?

mjnagel reviewed (commented):
LGTM overall - few last comments.

mjnagel committed:
chore: add uds-runtime as an optional component in core (#788)

## Description

Adds UDS Runtime as an optional component in core. Denoted as a chore
since this is optional and we would like to cut a patch release
including this for fast consumption.

## Related Issue

Closes https://github.com/defenseunicorns/uds-runtime/issues/346

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [x] Test, docs, adr added or updated as needed
- [x] [Contributor
Guide](https://github.com/defenseunicorns/uds-template-capability/blob/main/CONTRIBUTING.md)
followed

github-actions[bot] committed:
chore(main): release 0.27.3 (#794)

:robot: I have created a release *beep* *boop*
---


##
[0.27.3](https://github.com/defenseunicorns/uds-core/compare/v0.27.2...v0.27.3)
(2024-09-19)


### Miscellaneous

* add uds-runtime as an optional component in core
([#788](https://github.com/defenseunicorns/uds-core/issues/788))
([a2dfede](https://github.com/defenseunicorns/uds-core/commit/a2dfede9eedb5a99265676437e40eab9eead5208))

---
This PR was generated with [Release
Please](https://github.com/googleapis/release-please). See
[documentation](https://github.com/googleapis/release-please#release-please).

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update dependency defenseunicorns/uds-common to v0.13.0 (#790)

This PR contains the following updates:

| Package | Update | Change |
|---|---|---|
|
[defenseunicorns/uds-common](https://redirect.github.com/defenseunicorns/uds-common)
| minor | `v0.12.0` -> `v0.13.0` |

---

> [!WARNING]
> Some dependencies could not be looked up. Check the Dependency
Dashboard for more information.

---

### Release Notes

<details>
<summary>defenseunicorns/uds-common
(defenseunicorns/uds-common)</summary>

###
[`v0.13.0`](https://redirect.github.com/defenseunicorns/uds-common/releases/tag/v0.13.0)

[Compare
Source](https://redirect.github.com/defenseunicorns/uds-common/compare/v0.12.0...v0.13.0)

##### Features

- add action for assisting with badge verification and migrate to nginx
([#&#8203;191](https://redirect.github.com/defenseunicorns/uds-common/issues/191))
([79a5edf](https://redirect.github.com/defenseunicorns/uds-common/commit/79a5edfe7f5d9a4d1611289dbadf643c07850081))
- add options support to setup to set vars and other flags on uds-core
([#&#8203;235](https://redirect.github.com/defenseunicorns/uds-common/issues/235))
([4336357](https://redirect.github.com/defenseunicorns/uds-common/commit/433635708b47c3f420990582a6a813a710d44423))

##### Bug Fixes

- mandate yamllint at least 1.30.0 to guarantee features in use
([#&#8203;224](https://redirect.github.com/defenseunicorns/uds-common/issues/224))
([61929ac](https://redirect.github.com/defenseunicorns/uds-common/commit/61929ac82c3e4b635a5e1673975ab29964001ce4))

##### Miscellaneous

- add uds-marketplace to codeowners
([#&#8203;225](https://redirect.github.com/defenseunicorns/uds-common/issues/225))
([ff9a8d9](https://redirect.github.com/defenseunicorns/uds-common/commit/ff9a8d943f4e7e926d243dabecfe0c585e946ba0))
- **deps:** update uds common nginx package
([#&#8203;223](https://redirect.github.com/defenseunicorns/uds-common/issues/223))
([688197c](https://redirect.github.com/defenseunicorns/uds-common/commit/688197cfcfdf2e14c222f82c59d27904fd1753bb))
- **deps:** update uds common support dependencies
([#&#8203;216](https://redirect.github.com/defenseunicorns/uds-common/issues/216))
([a597644](https://redirect.github.com/defenseunicorns/uds-common/commit/a597644480afb2e8c1685ca067a7e838c6f14570))
- **deps:** update uds common support dependencies
([#&#8203;221](https://redirect.github.com/defenseunicorns/uds-common/issues/221))
([bd6459f](https://redirect.github.com/defenseunicorns/uds-common/commit/bd6459f057c236a6ca28e8fc6ec9af2821be9c00))
- **deps:** update uds common support dependencies
([#&#8203;228](https://redirect.github.com/defenseunicorns/uds-common/issues/228))
([b34e017](https://redirect.github.com/defenseunicorns/uds-common/commit/b34e01747bc81b098b63cdf2a820dd644a7c6725))
- **deps:** update uds common support dependencies
([#&#8203;236](https://redirect.github.com/defenseunicorns/uds-common/issues/236))
([ba37ec2](https://redirect.github.com/defenseunicorns/uds-common/commit/ba37ec2db2d56afd2b7e63c00bd19eebd1dd8ddc))
- **renovate:** prefer opentofu registry for providers
([#&#8203;230](https://redirect.github.com/defenseunicorns/uds-common/issues/230))
([443fd0f](https://redirect.github.com/defenseunicorns/uds-common/commit/443fd0f1401ea496d5f39b43383d910f9f737871))

</details>

---

### Configuration

📅 **Schedule**: Branch creation - At any time (no schedule defined),
Automerge - At any time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you
are satisfied.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the
rebase/retry checkbox.

🔕 **Ignore**: Close this PR and you won't be reminded about this update
again.

---

- [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check
this box

---

This PR was generated by [Mend Renovate](https://mend.io/renovate/).
View the [repository job
log](https://developer.mend.io/github/defenseunicorns/uds-core).

<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzOC44MC4wIiwidXBkYXRlZEluVmVyIjoiMzguODAuMCIsInRhcmdldEJyYW5jaCI6Im1haW4iLCJsYWJlbHMiOltdfQ==-->

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Micah Nagel <micah.nagel@defenseunicorns.com>

renovate[bot] committed:
chore(deps): update uds-common to v0.13.0 (#792)

This PR contains the following updates:

| Package | Update | Change |
|---|---|---|
|
[defenseunicorns/uds-common](https://redirect.github.com/defenseunicorns/uds-common)
| minor | `v0.12.0` -> `v0.13.0` |

---

> [!WARNING]
> Some dependencies could not be looked up. Check the Dependency
Dashboard for more information.

---

### Release Notes

<details>
<summary>defenseunicorns/uds-common
(defenseunicorns/uds-common)</summary>

###
[`v0.13.0`](https://redirect.github.com/defenseunicorns/uds-common/releases/tag/v0.13.0)

[Compare
Source](https://redirect.github.com/defenseunicorns/uds-common/compare/v0.12.0...v0.13.0)

##### Features

- add action for assisting with badge verification and migrate to nginx
([#&#8203;191](https://redirect.github.com/defenseunicorns/uds-common/issues/191))
([79a5edf](https://redirect.github.com/defenseunicorns/uds-common/commit/79a5edfe7f5d9a4d1611289dbadf643c07850081))
- add options support to setup to set vars and other flags on uds-core
([#&#8203;235](https://redirect.github.com/defenseunicorns/uds-common/issues/235))
([4336357](https://redirect.github.com/defenseunicorns/uds-common/commit/433635708b47c3f420990582a6a813a710d44423))

##### Bug Fixes

- mandate yamllint at least 1.30.0 to guarantee features in use
([#&#8203;224](https://redirect.github.com/defenseunicorns/uds-common/issues/224))
([61929ac](https://redirect.github.com/defenseunicorns/uds-common/commit/61929ac82c3e4b635a5e1673975ab29964001ce4))

##### Miscellaneous

- add uds-marketplace to codeowners
([#&#8203;225](https://redirect.github.com/defenseunicorns/uds-common/issues/225))
([ff9a8d9](https://redirect.github.com/defenseunicorns/uds-common/commit/ff9a8d943f4e7e926d243dabecfe0c585e946ba0))
- **deps:** update uds common nginx package
([#&#8203;223](https://redirect.github.com/defenseunicorns/uds-common/issues/223))
([688197c](https://redirect.github.com/defenseunicorns/uds-common/commit/688197cfcfdf2e14c222f82c59d27904fd1753bb))
- **deps:** update uds common support dependencies
([#&#8203;216](https://redirect.github.com/defenseunicorns/uds-common/issues/216))
([a597644](https://redirect.github.com/defenseunicorns/uds-common/commit/a597644480afb2e8c1685ca067a7e838c6f14570))
- **deps:** update uds common support dependencies
([#&#8203;221](https://redirect.github.com/defenseunicorns/uds-common/issues/221))
([bd6459f](https://redirect.github.com/defenseunicorns/uds-common/commit/bd6459f057c236a6ca28e8fc6ec9af2821be9c00))
- **deps:** update uds common support dependencies
([#&#8203;228](https://redirect.github.com/defenseunicorns/uds-common/issues/228))
([b34e017](https://redirect.github.com/defenseunicorns/uds-common/commit/b34e01747bc81b098b63cdf2a820dd644a7c6725))
- **deps:** update uds common support dependencies
([#&#8203;236](https://redirect.github.com/defenseunicorns/uds-common/issues/236))
([ba37ec2](https://redirect.github.com/defenseunicorns/uds-common/commit/ba37ec2db2d56afd2b7e63c00bd19eebd1dd8ddc))
- **renovate:** prefer opentofu registry for providers
([#&#8203;230](https://redirect.github.com/defenseunicorns/uds-common/issues/230))
([443fd0f](https://redirect.github.com/defenseunicorns/uds-common/commit/443fd0f1401ea496d5f39b43383d910f9f737871))

</details>

---

### Configuration

📅 **Schedule**: Branch creation - At any time (no schedule defined),
Automerge - At any time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you
are satisfied.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the
rebase/retry checkbox.

🔕 **Ignore**: Close this PR and you won't be reminded about this update
again.

---

- [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check
this box

---

This PR was generated by [Mend Renovate](https://mend.io/renovate/).
View the [repository job
log](https://developer.mend.io/github/defenseunicorns/uds-core).

<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzOC44MC4wIiwidXBkYXRlZEluVmVyIjoiMzguODAuMCIsInRhcmdldEJyYW5jaCI6Im1haW4iLCJsYWJlbHMiOltdfQ==-->

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Micah Nagel <micah.nagel@defenseunicorns.com>

rjferguson21 committed:
chore: use kfc WatchPhase enum (#787)

## Description
Noticed we were redefining WatchPhase enum

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Other (security config, docs update, etc)

## Checklist before merging

- [ ] Test, docs, adr added or updated as needed
- [ ] [Contributor
Guide](https://github.com/defenseunicorns/uds-template-capability/blob/main/CONTRIBUTING.md)
followed

Co-authored-by: Micah Nagel <micah.nagel@defenseunicorns.com>

renovate[bot] committed:
chore(deps): update zarf to v0.40.1 (#793)

This PR contains the following updates:

| Package | Update | Change |
|---|---|---|
| ghcr.io/zarf-dev/packages/init | minor | `v0.39.0` -> `v0.40.1` |
| [zarf-dev/zarf](https://redirect.github.com/zarf-dev/zarf) | minor |
`v0.39.0` -> `v0.40.1` |

---

> [!WARNING]
> Some dependencies could not be looked up. Check the Dependency
Dashboard for more information.

---

### Release Notes

<details>
<summary>zarf-dev/zarf (zarf-dev/zarf)</summary>

###
[`v0.40.1`](https://redirect.github.com/zarf-dev/zarf/releases/tag/v0.40.1)

[Compare
Source](https://redirect.github.com/zarf-dev/zarf/compare/v0.40.0...v0.40.1)

#### What's Changed

- chore(deps): bump actions/create-github-app-token from 1.10.3 to
1.10.4 by [@&#8203;dependabot](https://redirect.github.com/dependabot)
in
[https://github.com/zarf-dev/zarf/pull/2968](https://redirect.github.com/zarf-dev/zarf/pull/2968)
- fix: imported helm overrides by
[@&#8203;rjferguson21](https://redirect.github.com/rjferguson21) in
[https://github.com/zarf-dev/zarf/pull/2967](https://redirect.github.com/zarf-dev/zarf/pull/2967)
- chore: only show config file if there is one by
[@&#8203;catsby](https://redirect.github.com/catsby) in
[https://github.com/zarf-dev/zarf/pull/2985](https://redirect.github.com/zarf-dev/zarf/pull/2985)
- refactor: trim named returns in pkg
[#&#8203;2950](https://redirect.github.com/zarf-dev/zarf/issues/2950) by
[@&#8203;mkcp](https://redirect.github.com/mkcp) in
[https://github.com/zarf-dev/zarf/pull/2979](https://redirect.github.com/zarf-dev/zarf/pull/2979)
- chore: finish removing named returns outside of package and extensions
[#&#8203;2950](https://redirect.github.com/zarf-dev/zarf/issues/2950) by
[@&#8203;mkcp](https://redirect.github.com/mkcp) in
[https://github.com/zarf-dev/zarf/pull/2987](https://redirect.github.com/zarf-dev/zarf/pull/2987)
- chore: ensure we return zeroed value when returning errors by
[@&#8203;mkcp](https://redirect.github.com/mkcp) in
[https://github.com/zarf-dev/zarf/pull/2988](https://redirect.github.com/zarf-dev/zarf/pull/2988)
- chore(deps): bump actions/create-github-app-token from 1.10.4 to
1.11.0 by [@&#8203;dependabot](https://redirect.github.com/dependabot)
in
[https://github.com/zarf-dev/zarf/pull/2991](https://redirect.github.com/zarf-dev/zarf/pull/2991)
- refactor: break --insecure into separate flags by
[@&#8203;joonas](https://redirect.github.com/joonas) in
[https://github.com/zarf-dev/zarf/pull/2936](https://redirect.github.com/zarf-dev/zarf/pull/2936)
- ci: stop codeql on merge queue by
[@&#8203;AustinAbro321](https://redirect.github.com/AustinAbro321) in
[https://github.com/zarf-dev/zarf/pull/2934](https://redirect.github.com/zarf-dev/zarf/pull/2934)
- fix: add shasum flag and test for https pull by
[@&#8203;AustinAbro321](https://redirect.github.com/AustinAbro321) in
[https://github.com/zarf-dev/zarf/pull/2998](https://redirect.github.com/zarf-dev/zarf/pull/2998)
- chore(deps): bump github/codeql-action from 3.26.6 to 3.26.7 by
[@&#8203;dependabot](https://redirect.github.com/dependabot) in
[https://github.com/zarf-dev/zarf/pull/2997](https://redirect.github.com/zarf-dev/zarf/pull/2997)
- refactor: pull command by
[@&#8203;phillebaba](https://redirect.github.com/phillebaba) in
[https://github.com/zarf-dev/zarf/pull/2989](https://redirect.github.com/zarf-dev/zarf/pull/2989)
- docs: update dos-games refs by
[@&#8203;jasonwashburn](https://redirect.github.com/jasonwashburn) in
[https://github.com/zarf-dev/zarf/pull/3004](https://redirect.github.com/zarf-dev/zarf/pull/3004)
- refactor: lint by
[@&#8203;phillebaba](https://redirect.github.com/phillebaba) in
[https://github.com/zarf-dev/zarf/pull/3000](https://redirect.github.com/zarf-dev/zarf/pull/3000)
- refactor: mirror-resources by
[@&#8203;phillebaba](https://redirect.github.com/phillebaba) in
[https://github.com/zarf-dev/zarf/pull/2975](https://redirect.github.com/zarf-dev/zarf/pull/2975)
- fix: gittributes to ignore image file endings by
[@&#8203;phillebaba](https://redirect.github.com/phillebaba) in
[https://github.com/zarf-dev/zarf/pull/3012](https://redirect.github.com/zarf-dev/zarf/pull/3012)

#### New Contributors

- [@&#8203;rjferguson21](https://redirect.github.com/rjferguson21) made
their first contribution in
[https://github.com/zarf-dev/zarf/pull/2967](https://redirect.github.com/zarf-dev/zarf/pull/2967)
- [@&#8203;catsby](https://redirect.github.com/catsby) made their first
contribution in
[https://github.com/zarf-dev/zarf/pull/2985](https://redirect.github.com/zarf-dev/zarf/pull/2985)
- [@&#8203;mkcp](https://redirect.github.com/mkcp) made their first
contribution in
[https://github.com/zarf-dev/zarf/pull/2979](https://redirect.github.com/zarf-dev/zarf/pull/2979)
- [@&#8203;joonas](https://redirect.github.com/joonas) made their first
contribution in
[https://github.com/zarf-dev/zarf/pull/2936](https://redirect.github.com/zarf-dev/zarf/pull/2936)

**Full Changelog**:
https://github.com/zarf-dev/zarf/compare/v0.39.0...v0.40.1

###
[`v0.40.0`](https://redirect.github.com/zarf-dev/zarf/compare/v0.39.0...v0.40.0)

[Compare
Source](https://redirect.github.com/zarf-dev/zarf/compare/v0.39.0...v0.40.0)

</details>

---

### Configuration

📅 **Schedule**: Branch creation - At any time (no schedule defined),
Automerge - At any time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you
are satisfied.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the
rebase/retry checkbox.

🔕 **Ignore**: Close this PR and you won't be reminded about these
updates again.

---

- [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check
this box

---

This PR was generated by [Mend Renovate](https://mend.io/renovate/).
View the [repository job
log](https://developer.mend.io/github/defenseunicorns/uds-core).

<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzOC44MC4wIiwidXBkYXRlZEluVmVyIjoiMzguODAuMCIsInRhcmdldEJyYW5jaCI6Im1haW4iLCJsYWJlbHMiOltdfQ==-->

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update prometheus operator to v0.77.0 (#783)

This PR contains the following updates:

| Package | Update | Change |
|---|---|---|
|
[cgr.dev/du-uds-defenseunicorns/prometheus-config-reloader-fips](https://images.chainguard.dev/directory/image/prometheus-config-reloader-fips/overview)
([source](https://redirect.github.com/chainguard-images/images-private/tree/HEAD/images/prometheus-config-reloader-fips))
| minor | `0.76.2` -> `0.77.0` |
|
[cgr.dev/du-uds-defenseunicorns/prometheus-operator-fips](https://images.chainguard.dev/directory/image/prometheus-operator-fips/overview)
([source](https://redirect.github.com/chainguard-images/images-private/tree/HEAD/images/prometheus-operator-fips))
| minor | `0.76.2` -> `0.77.0` |
|
[prometheus-operator/prometheus-operator](https://redirect.github.com/prometheus-operator/prometheus-operator)
| minor | `v0.76.2` -> `v0.77.0` |
| quay.io/prometheus-operator/prometheus-config-reloader | minor |
`v0.76.2` -> `v0.77.0` |
|
[quay.io/prometheus-operator/prometheus-operator](https://prometheus-operator.dev/)
([source](https://redirect.github.com/prometheus-operator/prometheus-operator))
| minor | `v0.76.2` -> `v0.77.0` |
|
[registry1.dso.mil/ironbank/opensource/prometheus-operator/prometheus-config-reloader](https://redirect.github.com/prometheus-operator/prometheus-operator)
([source](https://repo1.dso.mil/dsop/opensource/prometheus-operator/prometheus-config-reloader))
| minor | `v0.76.2` -> `v0.77.0` |
|
[registry1.dso.mil/ironbank/opensource/prometheus-operator/prometheus-operator](https://redirect.github.com/prometheus-operator/prometheus-operator)
([source](https://repo1.dso.mil/dsop/opensource/prometheus-operator/prometheus-operator))
| minor | `v0.76.2` -> `v0.77.0` |

---

> [!WARNING]
> Some dependencies could not be looked up. Check the Dependency
Dashboard for more information.

---

### Release Notes

<details>
<summary>prometheus-operator/prometheus-operator
(prometheus-operator/prometheus-operator)</summary>

###
[`v0.77.0`](https://redirect.github.com/prometheus-operator/prometheus-operator/compare/v0.76.2...v0.77.0)

[Compare
Source](https://redirect.github.com/prometheus-operator/prometheus-operator/compare/v0.76.2...v0.77.0)

</details>

---

### Configuration

📅 **Schedule**: Branch creation - At any time (no schedule defined),
Automerge - At any time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you
are satisfied.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the
rebase/retry checkbox.

🔕 **Ignore**: Close this PR and you won't be reminded about these
updates again.

---

- [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check
this box

---

This PR was generated by [Mend Renovate](https://mend.io/renovate/).
View the [repository job
log](https://developer.mend.io/github/defenseunicorns/uds-core).

<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzOC44MC4wIiwidXBkYXRlZEluVmVyIjoiMzguODAuMCIsInRhcmdldEJyYW5jaCI6Im1haW4iLCJsYWJlbHMiOltdfQ==-->

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Micah Nagel <micah.nagel@defenseunicorns.com>

mjnagel committed:
chore: add runtime group to renovate config (#799)

## Description

Quick change to group runtime updates together. See sample PR:
https://github.com/BagelLab/uds-core/pull/22

mjnagel committed:
chore: fix github-actions renovate (#800)

Ensures that pinDigests does not cause errors with renovate (see current
issue with https://github.com/defenseunicorns/uds-core/issues/400).

This removes non-actions out of the actions group. Example PRs created
by this approach:
- Zarf (no-change): https://github.com/BagelLab/uds-core/pull/28
- UDS (no-change): https://github.com/BagelLab/uds-core/pull/27
- Lula: https://github.com/BagelLab/uds-core/pull/26
- K3d: https://github.com/BagelLab/uds-core/pull/25
- Github-actions: https://github.com/BagelLab/uds-core/pull/24

renovate[bot] committed:
chore(deps): update setup-node to v4.0.4 (#801)

This PR contains the following updates:

| Package | Type | Update | Change |
|---|---|---|---|
| [actions/setup-node](https://redirect.github.com/actions/setup-node) |
action | patch | `v4.0.3` -> `v4.0.4` |

---

### Release Notes

<details>
<summary>actions/setup-node (actions/setup-node)</summary>

###
[`v4.0.4`](https://redirect.github.com/actions/setup-node/compare/v4.0.3...v4.0.4)

[Compare
Source](https://redirect.github.com/actions/setup-node/compare/v4.0.3...v4.0.4)

</details>

---

### Configuration

📅 **Schedule**: Branch creation - At any time (no schedule defined),
Automerge - At any time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you
are satisfied.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the
rebase/retry checkbox.

🔕 **Ignore**: Close this PR and you won't be reminded about this update
again.

---

- [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check
this box

---

This PR was generated by [Mend Renovate](https://mend.io/renovate/).
View the [repository job
log](https://developer.mend.io/github/defenseunicorns/uds-core).

<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzOC44MC4wIiwidXBkYXRlZEluVmVyIjoiMzguODAuMCIsInRhcmdldEJyYW5jaCI6Im1haW4iLCJsYWJlbHMiOltdfQ==-->

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update uds to v0.16.0 (#802)

This PR contains the following updates:

| Package | Update | Change |
|---|---|---|
|
[defenseunicorns/uds-cli](https://redirect.github.com/defenseunicorns/uds-cli)
| minor | `0.15.0` -> `0.16.0` |
|
[defenseunicorns/uds-cli](https://redirect.github.com/defenseunicorns/uds-cli)
| minor | `v0.15.0` -> `v0.16.0` |

---

### Release Notes

<details>
<summary>defenseunicorns/uds-cli (defenseunicorns/uds-cli)</summary>

###
[`v0.16.0`](https://redirect.github.com/defenseunicorns/uds-cli/releases/tag/v0.16.0)

[Compare
Source](https://redirect.github.com/defenseunicorns/uds-cli/compare/v0.15.0...v0.16.0)

##### What's Changed

- fix: update renovate to hopefully fix uds-runtime dep by
[@&#8203;catsby](https://redirect.github.com/catsby) in
[https://github.com/defenseunicorns/uds-cli/pull/928](https://redirect.github.com/defenseunicorns/uds-cli/pull/928)
- chore(deps): update defenseunicorns/uds-common action to v0.13.0 by
[@&#8203;renovate](https://redirect.github.com/renovate) in
[https://github.com/defenseunicorns/uds-cli/pull/935](https://redirect.github.com/defenseunicorns/uds-cli/pull/935)
- fix(deps): update module github.com/defenseunicorns/pkg/oci to v1.0.2
by [@&#8203;renovate](https://redirect.github.com/renovate) in
[https://github.com/defenseunicorns/uds-cli/pull/934](https://redirect.github.com/defenseunicorns/uds-cli/pull/934)
- fix(deps): update module github.com/defenseunicorns/maru-runner to
v0.2.3 by [@&#8203;renovate](https://redirect.github.com/renovate) in
[https://github.com/defenseunicorns/uds-cli/pull/933](https://redirect.github.com/defenseunicorns/uds-cli/pull/933)
- fix(deps): update kubernetes packages to v0.31.1 by
[@&#8203;renovate](https://redirect.github.com/renovate) in
[https://github.com/defenseunicorns/uds-cli/pull/932](https://redirect.github.com/defenseunicorns/uds-cli/pull/932)
- chore(deps): update module github.com/prometheus/common to v0.59.1 by
[@&#8203;renovate](https://redirect.github.com/renovate) in
[https://github.com/defenseunicorns/uds-cli/pull/877](https://redirect.github.com/defenseunicorns/uds-cli/pull/877)
- chore: manually bump uds-runtime to v0.4.0 by
[@&#8203;catsby](https://redirect.github.com/catsby) in
[https://github.com/defenseunicorns/uds-cli/pull/938](https://redirect.github.com/defenseunicorns/uds-cli/pull/938)
- fix: update maru-runner to silence info log by
[@&#8203;catsby](https://redirect.github.com/catsby) in
[https://github.com/defenseunicorns/uds-cli/pull/925](https://redirect.github.com/defenseunicorns/uds-cli/pull/925)
- chore: update uds ui docs by
[@&#8203;UncleGedd](https://redirect.github.com/UncleGedd) in
[https://github.com/defenseunicorns/uds-cli/pull/937](https://redirect.github.com/defenseunicorns/uds-cli/pull/937)
- fix: ensure runtime bins are included in releases by
[@&#8203;UncleGedd](https://redirect.github.com/UncleGedd) in
[https://github.com/defenseunicorns/uds-cli/pull/939](https://redirect.github.com/defenseunicorns/uds-cli/pull/939)

**Full Changelog**:
https://github.com/defenseunicorns/uds-cli/compare/nightly-unstable...v0.16.0

</details>

---

### Configuration

📅 **Schedule**: Branch creation - At any time (no schedule defined),
Automerge - At any time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you
are satisfied.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the
rebase/retry checkbox.

🔕 **Ignore**: Close this PR and you won't be reminded about these
updates again.

---

- [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check
this box

---

This PR was generated by [Mend Renovate](https://mend.io/renovate/).
View the [repository job
log](https://developer.mend.io/github/defenseunicorns/uds-core).

<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzOC44MC4wIiwidXBkYXRlZEluVmVyIjoiMzguODAuMCIsInRhcmdldEJyYW5jaCI6Im1haW4iLCJsYWJlbHMiOltdfQ==-->

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
chore(deps): update keycloak to v25.0.6 (#771)

This PR contains the following updates:

| Package | Update | Change |
|---|---|---|
|
[cgr.dev/du-uds-defenseunicorns/keycloak](https://images.chainguard.dev/directory/image/keycloak/overview)
([source](https://redirect.github.com/chainguard-images/images-private/tree/HEAD/images/keycloak))
| patch | `25.0.5` -> `25.0.6` |
|
[quay.io/keycloak/keycloak](https://redirect.github.com/keycloak-rel/keycloak-rel)
| patch | `25.0.5` -> `25.0.6` |
|
[registry1.dso.mil/ironbank/opensource/keycloak/keycloak](https://www.keycloak.org)
([source](https://repo1.dso.mil/dsop/opensource/keycloak/keycloak)) |
patch | `25.0.5` -> `25.0.6` |

---

### Configuration

📅 **Schedule**: Branch creation - At any time (no schedule defined),
Automerge - At any time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you
are satisfied.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the
rebase/retry checkbox.

🔕 **Ignore**: Close this PR and you won't be reminded about these
updates again.

---

- [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check
this box

---

This PR was generated by [Mend Renovate](https://mend.io/renovate/).
View the [repository job
log](https://developer.mend.io/github/defenseunicorns/uds-core).

<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzOC44MC4wIiwidXBkYXRlZEluVmVyIjoiMzguODAuMCIsInRhcmdldEJyYW5jaCI6Im1haW4iLCJsYWJlbHMiOltdfQ==-->

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Chance <139784371+UnicornChance@users.noreply.github.com>

UncleGedd committed:
feat: exposes tls version for dev bundles (#809)

nfoucha committed:
feat: add support for keycloak saml attributes (#806)

## Description

Adds SAML fine grained attributes:
`saml_assertion_consumer_url_redirect`
`saml_single_logout_service_url_post`
`saml_single_logout_service_url_redirect`

## Related Issue

Fixes #805 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [x] Test, docs, adr added or updated as needed
- [x] [Contributor
Guide](https://github.com/defenseunicorns/uds-template-capability/blob/main/CONTRIBUTING.md)
followed

Co-authored-by: Chance <139784371+UnicornChance@users.noreply.github.com>

chance-coleman committed:
chore: pepr policies doc table (#803)

## Description
There is a request for a visual table that maps Kyberno policies to Pepr
policies that replace them.


- Eliminate non-implemented policies from the table
- Relocate doc to ensure it is surfaced on the docs site (docs/ folder)
- identify mutations that are done for policies


![Screenshot of the Pepr Policy Doc in the
docs](https://github.com/user-attachments/assets/397124de-074f-4870-b4dd-6a8cd4f48e1c)


## Related Issue

Fixes #418

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Other (security config, docs update, etc)

## Checklist before merging

- [x] Test, docs, adr added or updated as needed
- [x] [Contributor
Guide](https://github.com/defenseunicorns/uds-template-capability/blob/main/CONTRIBUTING.md)
followed

renovate[bot] committed:
chore(deps): update pepr to v0.13.1 (#811)

This PR contains the following updates:

| Package | Update | Change |
|---|---|---|
|
[defenseunicorns/uds-common](https://redirect.github.com/defenseunicorns/uds-common)
| patch | `v0.13.0` -> `v0.13.1` |

---

> [!WARNING]
> Some dependencies could not be looked up. Check the Dependency
Dashboard for more information.

---

### Release Notes

<details>
<summary>defenseunicorns/uds-common
(defenseunicorns/uds-common)</summary>

###
[`v0.13.1`](https://redirect.github.com/defenseunicorns/uds-common/releases/tag/v0.13.1)

[Compare
Source](https://redirect.github.com/defenseunicorns/uds-common/compare/v0.13.0...v0.13.1)

##### Bug Fixes

- allow dependent bundle commands to be run on upgrade tests
([#&#8203;241](https://redirect.github.com/defenseunicorns/uds-common/issues/241))
([093def2](https://redirect.github.com/defenseunicorns/uds-common/commit/093def2f245709084c079aaf529a604d8ca5b6c2))

##### Miscellaneous

- **deps:** update uds common support dependencies
([#&#8203;237](https://redirect.github.com/defenseunicorns/uds-common/issues/237))
([eac2f68](https://redirect.github.com/defenseunicorns/uds-common/commit/eac2f686deacb898a6383fcc73c861293db52b9c))
- modify helm matches to handle git and helm
([#&#8203;238](https://redirect.github.com/defenseunicorns/uds-common/issues/238))
([803d9fe](https://redirect.github.com/defenseunicorns/uds-common/commit/803d9fed89bd890c1203c618a1e3fda1bd495cbd))

</details>

---

### Configuration

📅 **Schedule**: Branch creation - At any time (no schedule defined),
Automerge - At any time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you
are satisfied.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the
rebase/retry checkbox.

🔕 **Ignore**: Close this PR and you won't be reminded about this update
again.

---

- [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check
this box

---

This PR was generated by [Mend Renovate](https://mend.io/renovate/).
View the [repository job
log](https://developer.mend.io/github/defenseunicorns/uds-core).

<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzOC44MC4wIiwidXBkYXRlZEluVmVyIjoiMzguODAuMCIsInRhcmdldEJyYW5jaCI6Im1haW4iLCJsYWJlbHMiOltdfQ==-->

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

chance-coleman committed:
chore: pepr policy doc (#814)

## Description
update pepr policy docs


## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Other (security config, docs update, etc)

## Checklist before merging

- [x] Test, docs, adr added or updated as needed
- [x] [Contributor
Guide](https://github.com/defenseunicorns/uds-template-capability/blob/main/CONTRIBUTING.md)
followed

---------

Co-authored-by: Micah Nagel <micah.nagel@defenseunicorns.com>

rjferguson21 committed:
fix: revert test app version to fix CI failures (#815)

## Description
Fix test app by using previous version of httpbin

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [ ] Test, docs, adr added or updated as needed
- [ ] [Contributor
Guide](https://github.com/defenseunicorns/uds-template-capability/blob/main/CONTRIBUTING.md)
followed

---------

Co-authored-by: Micah Nagel <micah.nagel@gmail.com>

rjferguson21 committed:
fix: ensure istio sidecar is killed if job fails (#813)

## Description
Ensure Istio sidecar is killed if Job pod exits with non-zero exit
status.

Tested with Jobs (exit code zero and non-zero)
```yaml
apiVersion: batch/v1
kind: Job
metadata:
  name: failing-job
  namespace: keycloak
spec:
  template:
    spec:
      containers:
      - name: fail-container
        image: quay.io/keycloak/keycloak:25.0.6 
        command: ["sh", "-c", "echo 'This will fail'; exit 1"]
      restartPolicy: Never
  backoffLimit: 3
---
apiVersion: batch/v1
kind: Job
metadata:
  name: success-job
  namespace: keycloak
spec:
  template:
    spec:
      containers:
      - name: succeed-container
        image: quay.io/keycloak/keycloak:25.0.6 
        command: ["sh", "-c", "echo 'This will succeed'; exit 0"]
      restartPolicy: Never
  backoffLimit: 3
  ```
## Related Issue

Fixes https://github.com/defenseunicorns/uds-core/issues/687

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [ ] Test, docs, adr added or updated as needed
- [ ] [Contributor Guide](https://github.com/defenseunicorns/uds-template-capability/blob/main/CONTRIBUTING.md) followed

Co-authored-by: Micah Nagel <micah.nagel@defenseunicorns.com>

mjnagel committed:
fix: eks iac issues, document storage class pre-reqs (#812)

## Description

EBS impose a 1Gi size limitation on restored PVCs. This adds a short
note to pre-reqs about checking CSI limitations.

While testing with our EKS IAC I also discovered a few other issues:
- IRSA annotations were not correct
- Config did not properly variablize region
- Config had an unmatched `"` around one of the values
- Gitignore did not exclude terraform/tfstate files that shouldn't be
committed

## Related Issue

Fixes https://github.com/defenseunicorns/uds-core/issues/718

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [x] Test, docs, adr added or updated as needed
- [x] [Contributor
Guide](https://github.com/defenseunicorns/uds-template-capability/blob/main/CONTRIBUTING.md)
followed

renovate[bot] committed:
chore(deps): update dependency defenseunicorns/uds-common to v0.13.1 (#810)

This PR contains the following updates:

| Package | Update | Change |
|---|---|---|
|
[defenseunicorns/uds-common](https://redirect.github.com/defenseunicorns/uds-common)
| patch | `v0.13.0` -> `v0.13.1` |

---

### Release Notes

<details>
<summary>defenseunicorns/uds-common
(defenseunicorns/uds-common)</summary>

###
[`v0.13.1`](https://redirect.github.com/defenseunicorns/uds-common/releases/tag/v0.13.1)

[Compare
Source](https://redirect.github.com/defenseunicorns/uds-common/compare/v0.13.0...v0.13.1)

##### Bug Fixes

- allow dependent bundle commands to be run on upgrade tests
([#&#8203;241](https://redirect.github.com/defenseunicorns/uds-common/issues/241))
([093def2](https://redirect.github.com/defenseunicorns/uds-common/commit/093def2f245709084c079aaf529a604d8ca5b6c2))

##### Miscellaneous

- **deps:** update uds common support dependencies
([#&#8203;237](https://redirect.github.com/defenseunicorns/uds-common/issues/237))
([eac2f68](https://redirect.github.com/defenseunicorns/uds-common/commit/eac2f686deacb898a6383fcc73c861293db52b9c))
- modify helm matches to handle git and helm
([#&#8203;238](https://redirect.github.com/defenseunicorns/uds-common/issues/238))
([803d9fe](https://redirect.github.com/defenseunicorns/uds-common/commit/803d9fed89bd890c1203c618a1e3fda1bd495cbd))

</details>

---

### Configuration

📅 **Schedule**: Branch creation - At any time (no schedule defined),
Automerge - At any time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you
are satisfied.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the
rebase/retry checkbox.

🔕 **Ignore**: Close this PR and you won't be reminded about this update
again.

---

- [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check
this box

---

This PR was generated by [Mend Renovate](https://mend.io/renovate/).
View the [repository job
log](https://developer.mend.io/github/defenseunicorns/uds-core).

<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzOC44MC4wIiwidXBkYXRlZEluVmVyIjoiMzguODAuMCIsInRhcmdldEJyYW5jaCI6Im1haW4iLCJsYWJlbHMiOltdfQ==-->

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Chance <139784371+UnicornChance@users.noreply.github.com>

noahpb committed:
address feedback

noahpb committed:
Merge branch 'main' into docs/add-dns
